Source/Csla/Core/BusinessBase.cs

```diff
@@ -1227,6 +1227,33 @@ public virtual BrokenRulesCollection BrokenRulesCollection
 
     #region Data Access
 
+    /// <summary>
+    /// Await this method to ensure business object is not busy.
+    /// </summary>
+    public async Task WaitForIdle()
+    {
+      var cslaOptions = ApplicationContext.GetRequiredService<Configuration.CslaOptions>();
+      await WaitForIdle(TimeSpan.FromSeconds(cslaOptions.DefaultWaitForIdleTimeoutInSeconds)).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Await this method to ensure business object is not busy.
+    /// </summary>
+    /// <param name="timeout">Timeout duration</param>
+    public virtual Task WaitForIdle(TimeSpan timeout)
```

Comment:
Is there a way to have just one virtual overload?

I've done this before, with two virtual overloads that "do the same thing" and it becomes a serious point of confusion and bugs for people who overload the method(s) over time.

Reply:
Yeah. I'll think about a proper solution :)